navigation/ios/NavigationState.swift

```diff
@@ -23,19 +23,22 @@ class NavigationState<
     private let buildNavigation: (([Route], Route) -> PilotNavigationType<ScreenData, NavModifier>?)?
     private let navigationManager: PilotNavigationManager<Route, Action>?
     private let actionListener: ActionListener<Route, Action>
+    private let popRoot: (() -> Void)?
```

Comment:
I agree with the behavior, however I'm not sure about the naming, this feels inconsistant with the `handleAction` closure. Maybe `handlePopRoot` if that is a thing?

On android I would've gone for `onRootPopped` and `onAction` but whatever feels more natural in the iOS terms is fine with me.